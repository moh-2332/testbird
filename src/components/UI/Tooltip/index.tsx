import React from 'react';

import classes from "./Tooltip.module.scss"

const Tooltip: React.FC<{ text: string }> = ({ text }) => {
    return <span className={classes.tooltip}>{text}<i></i></span>;
}

export default Tooltip;