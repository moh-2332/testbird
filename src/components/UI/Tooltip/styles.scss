@import '../../../styles/colors';

.tooltip {
    min-width: 90px;
    top: -15px;
    left: 5%;
    transform: translate(-30%, -100%);
    padding: 7px;
    text-align: center;
    color: $dark;
    background-color: $light;
    font-weight: normal;
    font-size: 0.75rem;
    position: absolute;
    z-index: 1;
    box-sizing: border-box;
    box-shadow: 0 1px 8px rgb(0 0 0 / 20%);
    
    i {
        position:absolute;
        top:100%;
        left:50%;
        margin-left:-15px;
        width:30px;
        height:15px;
        overflow:hidden;
    }

    i::after {
        content:'';
        position:absolute;
        width:15px;
        height:15px;
        left:50%;
        transform:translate(-50%,-50%) rotate(45deg);
        background-color: $light;
        box-shadow: 0 1px 8px rgb(0 0 0 / 20%);
    }
}



