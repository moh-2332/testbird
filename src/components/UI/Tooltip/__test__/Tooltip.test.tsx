import React from "react";

import { screen, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";

import Tooltip from "../";

describe("Tooltip tests", () => {
    it("Should match the snapshot", () => {
        const tooltip = renderer.create(<Tooltip text="help" />);
        expect(tooltip).toMatchSnapshot();
    });

    it("Should renders correctly", () => {
        render(<Tooltip text="help" />);

        const sampleText = screen.getByText(/help/i);
        expect(sampleText).toBeInTheDocument();

        const tooltip = screen.getByTestId("tooltip");
        expect(tooltip).toBeInTheDocument();
    });
});