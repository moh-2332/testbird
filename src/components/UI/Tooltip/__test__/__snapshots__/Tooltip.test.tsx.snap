// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip tests Should match the snapshot 1`] = `
<span
  className="tooltip"
  data-testid="tooltip"
>
  help
  <i />
</span>
`;
